app/src/main/java/com/keylesspalace/tusky/TabData.kt

```diff
@@ -36,15 +45,25 @@ const val DIRECT = "Direct"
 const val TRENDING = "Trending"
 const val HASHTAG = "Hashtag"
 const val LIST = "List"
+const val EDIT_PROFILE = "Edit profile"
+const val FAVOURITES = "Favourites"
+const val BOOKMARKS = "Bookmarks"
+const val FOLLOW_REQUESTS = "Follow requests"
+const val LISTS = "Lists"
+const val DRAFTS = "Drafts"
+const val SCHEDULED_POSTS = "Scheduled posts"
+const val ANNOUNCEMENTS = "Announcements"
 
+@Parcelize
 data class TabData(
```

Comment:
This should definitely have a different name now, how about "AppScreen" or "ScreenData"?

Reply:
And I wonder if two classes would model things better here.
In the end, all screens can be shown in their own Activity, and some can also be shown as Tabs. So maybe like this

```kotlin
open class ScreenData(
    val id: String,
    @StringRes val text: Int,
    val icon: Icon,
    val arguments: List<String> = emptyList(),
    val intentAction: (Context, List<String>, Boolean) -> Intent,
    val title: (Context) -> String = { context -> context.getString(text) },
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ScreenData

        if (id != other.id) return false
        if (arguments != other.arguments) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + arguments.hashCode()
        return result
    }

    open fun withArguments(args: List<String>): ScreenData = ScreenData(
        id = id,
        text = text,
        icon = icon,
        arguments = args,
        intentAction = intentAction,
        title = title
    )
}


class TabScreenData(
    id: String,
    text: Int,
    icon: Icon,
    arguments: List<String> = emptyList(),
    val fragmentAction: (List<String>) -> Fragment,
    title: (Context) -> String = { context -> context.getString(text) },
) : ScreenData (
    id = id,
    text = text,
    icon = icon,
    arguments = arguments,
    intentAction = { context, tabData, accountLocked -> TabActivity.getIntent(context, id, accountLocked) },
    title = title
) {
    override fun withArguments(args: List<String>) = TabScreenData(
        id = id,
        text = text,
        icon = icon,
        arguments = args,
        fragmentAction = fragmentAction,
        title = title
    )
}
```

then have one `createTabScreenDataFromId(...): TabScreenData` and a `createScreenDataFromId(...): ScreenData` that calls the former, instead of `AllowedContext` check for the class, use `intentAction` whenever you need an intent, `fragmentAction` for Fragments, etc...

(But is works now and isn't super bad, better leave it if it is too complicated)

Reply:
Definitely, I like the idea of ScreenData. TabData really makes no sense any more in the given context. I'll see how easy the change to inheritance is.

Reply:
It took a bit to refactor it. There are a lot of files touched by this. I think it is better in the long run.

Reply:
@connyduck  Alright, the latest code changes should represent what you asked for.